### PR TITLE
Added AWS AppSync based APIs

### PR DESCRIPTION
**Pull request by giedri**
*Issue #, if available:*

*Description of changes:*
Added AWS AppSync based APIs with unit and integration testing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: serverless-graphql-api/javascript-appsync-sam-js/README.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+Unit tests use AWS SDK to evaluate AWS AppSync resolver code. They use mock data, not the integrations with the backend services. Use following command to run unit tests:
+```bash
+npm run test:unit
+```
+
+Integration tests send GraphQL queries to the AWS AppSync endpoint and verify responses received. In addition, integration tests use WebSocket connection to the AWS AppSync for GraphQL subscriptions. You should run the test initialization script first, to create necessary accounts in Amazon Cognito and retrieve access tokens. Also make sure to cleanup after integration testing is over:
+
+```bash
+eval "$(./__tests__/integration/test_init.sh)"
```
Comment (dancfox) [inline serverless-graphql-api/javascript-appsync-sam-js/README.md:76]: I get an error the first time I run this. Is that expected?

```
An error occurred (UserNotFoundException) when calling the AdminDeleteUser operation: User does not exist.
```

**Comment by giedri** [comment]
Paths fixed, tested, all requested fixed in place.

File: serverless-graphql-api/javascript-appsync-sam-js/README.md (hunk 1 above)
Comment (giedri) [reply serverless-graphql-api/javascript-appsync-sam-js/README.md]: Fixed

**Comment by giedri** [comment]
> I received the following error: Screenshot 2023-04-19 at 1 19 55 PM

You are copy/pasting command with the path. Drop "~/.../serverless-samples$" part from the command

**Comment by dancfox** [comment]
I made some edits to the README.

File: serverless-graphql-api/javascript-appsync-sam-js/README.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+javascript-appsync-sam-js:~$ git remote add origin <URL to AWS CodeCommit repository>
+javascript-appsync-sam-js:~$ git push origin main
+```
+
+Navigate to the AWS CodePipeline in the AWS Management Console and release this change if needed by clicking "Release change" button.
+
+![CodePipeline](./assets/CodePipeline.png)
+
+Note that the same Amazon Cognito stack is used in both testing and production deployment stages, and the same user credentials can be used for testing and API access.
+
```
Comment (dancfox) [inline serverless-graphql-api/javascript-appsync-sam-js/README.md:53]: Suggest you add some copy about the Manual Approval process.

Comment (dancfox) [reply serverless-graphql-api/javascript-appsync-sam-js/README.md:53]: Suggest you add some copy asking me to verify something. For example, when I see this thing, I'm done. Tell me what I've accomplished. Maybe suggest I inspect something to learn about what I've done. Congratulate me. Make sure I understand that I'm complete here and that the next part is an alternate pathway I can go down. Why might I want to go down that pathway?

File: serverless-graphql-api/shared/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+aws cognito-idp admin-confirm-sign-up --user-pool-id <cognito user pool id> --username <username> 
+```
+
+After this first step step your new user account will be able to access public data and create new bookings. To add locations and resources you will need to navigate to AWS Console, pick Amazon Cognito service, select User Pool instance that was created during this deployment, navigate to "Users and Groups", and add your user to administrative users group. 
+
+While using command line or third party tools such as Postman to test APIs, you will need to provide Identity Token in the request "Authorization" header. You can authenticate with Amazon Cognito User Pool using AWS CLI (this command is also available in SAM template outputs) and use IdToken value present in the output of the command (it is available in the stack outputs as well):
+
+```bash
+aws cognito-idp initiate-auth --auth-flow USER_PASSWORD_AUTH --client-id <cognito user pool application client id> --auth-parameters USERNAME=<username>,PASSWORD=<password>
```
Comment (dancfox) [inline serverless-graphql-api/shared/README.md:28]: This command provides me with an "AccessToken". How does this AccessToken fit into the paragraph above describing authorization?

**Comment by dancfox** [comment]
Made some changes to the shared README.

File: serverless-graphql-api/javascript-appsync-sam-js/README.md (hunk 2 above)
Comment (dancfox) [reply serverless-graphql-api/javascript-appsync-sam-js/README.md:53]: Could you make similar modifications to the README's in the two other directories if relevant?

Comment (giedri) [reply serverless-graphql-api/javascript-appsync-sam-js/README.md:82]: Updated instructions with walkthrough of what was created, screenshots, references to instructions on how to create API client, note on manual deployment.

**Comment by dancfox** [comment]
I think you overwrote the commits I made [here](https://github.com/aws-samples/serverless-samples/pull/71/commits/f1485461832b456df3f43bd694cc61885049c733).

File: serverless-graphql-api/assets/StackOutputs.png
[the remark this answers is not in the document]
Comment (giedri) [reply serverless-graphql-api/assets/StackOutputs.png:1]: Done

**Comment by dancfox** [comment]
Approved!

**Pull request merged**